sw/device/lib/flash_ctrl.h

```diff
@@ -117,6 +112,24 @@ void flash_default_region_access(bool rd_en, bool prog_en, bool erase_en);
  */
 void flash_cfg_region(const mp_region_t *region_cfg);
 
+/** Get number of flash banks */
+uint32_t flash_get_banks();
```

Comment:
You sure these don't want to be constants? E.g. `extern const uint32_t kWhatever` here, with a definition in the .c file?

Reply:
so these should be constants, but i wanted to generate them from the params in the hjson.
do we need to add to how those are generated?